CONTRIBUTING.md

```diff
@@ -3,10 +3,79 @@ There is a white-list for contributors who have signed the OCA so please add a c
+models the [review process][2] used by NetBeans. Its most important points are highlighted in this 
+document.
+
+An API isn't just what is written in Javadoc, it is not just about the [method signatures][4]. 
+The API includes properties, files, ports, protocols that are being read or written to.
+It includes L10N, packages, versioning. It includes runtime behavior, threading, 
+memory management, etc. In short API is *everything somebody else can depend on*.
+
+Trivial API Change
+------------------
+
+It is very likely that by changing something in the Truffle repository, you are changing an API. If you
+do so, it is important to stick with essential properties of a good API change:
+
+- be backward compatible
+- be well documented
+- have sufficient test coverage
+- be ready for future evolution
+
+If your change satisfies these properties you are eligible for a **fast-track** review: create a pull request,
+let people know about it, assign a responsible reviewer, react to *requests for change* and after a week 
+you should be ready to merge. Announce the *final 24 hour count down* and then merge or convince an
+approved committer to merge.
+
+Trivial change is usually about single method/class addition or other *little improvements* of existing APIs. 
+Additions of whole packages very likely fall into complex change category. In case of disagreement any reviewer
+can turn a trivial change into complex one by requesting **standard** review.
+
+Complex API Change
+------------------
+
+If an API change is inherently backward *incompatible*, or if it is huge, it is subject to **standard** review.
+The purpose of such review is to seek wider consensus among involved parties. The ideal state of such review
+should:
+
+- describe the intention, use-cases and let everyone comment
+- nominate four "voting" reviewers
+- have an *inception call* where reviewers approve/reject the project and set constraints/directions
+
+After this initial discussion the request has to be polished to address the result of the discussion. When done
+there should be another, *final review* where the "voting" reviewers decide whether to merge the change or not.
+
+**Standard** review should usually take longer than **fast-track**, but if there is a consensus of four reviewers
+and justification for such hurry (usually a release deadline), it can be finished in a day.
+
+No API Change
+-------------
+
+Adding tests verifying existing behavior, improving documentation to describe current behavior, refactoring 
+internal code without influencing functionality, mangling with private or package private members of API
+classes and other *internal changes* do not need an API review at all.
+
+It is still preferrable to designate additional reviewer to consider the intentions from a different 
+angle and merge the change. Alternatively one can treat these "no API changes" as trivial API changes
+and request **fast-track** review "just to be sure".
```

Comment:
This part isn't clear enough about if and how many reviewer such a change needs ("no API review at all", "*additional* reviewer"; if it's not fast-track by default, what is it then?).